api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -367,6 +368,35 @@ public Long findDomainCountByStandard(String domain, String term, Boolean standa
         : count;
   }
 
+  @Override
+  public List<CardCount> findDomainCounts(String term, Boolean standard, List<Domain> domains) {
+    List<String> strDomains =
+        domains.stream().map(d1 -> d1.toString()).collect(Collectors.toList());
+    List<CardCount> cardCounts =
+        cbCriteriaDao.findDomainCountsByCode(term, standard, strDomains).stream()
+            .filter(cardCount -> cardCount.getCount() > 0)
+            .map(cohortBuilderMapper::dbModelToClient)
+            .collect(Collectors.toList());
+    // filter strDomains to remove domains that have a cardCount
+    strDomains.removeAll(
+        cardCounts.stream().map(c -> c.getDomain().toString()).collect(Collectors.toList()));
+    // modify search term and call
+    cardCounts.addAll(
+        cbCriteriaDao.findDomainCounts(modifyTermMatch(term), standard, strDomains).stream()
+            .filter(cardCount -> cardCount.getCount() > 0)
+            .map(cohortBuilderMapper::dbModelToClient)
+            .collect(Collectors.toList()));
```

Comment:
The logic should be:
if findDomainCountsByCode returns data then only populate card counts with that
otherwise call findDomainCounts and use that data. 
We don't mix and match code matches with synonym matches

Reply:
Currently we do the --> getcountsByCode and if there are no counts then modify the search term and do a getcounts by standard and source  for *each domain*. We do the same by calling domain counts by code for *all* domain, retain thoes that have counts and then call domain counts by std/source for the *remaining* domain that do not have counts by modifying the search term